src/middlewared/middlewared/plugins/account.py

```diff
@@ -301,6 +307,13 @@ async def query(self, filters, options):
             else:
                 entry.update({'nt_name': None, 'sid': None})
 
+            user_roles = []
+
+            for group_pk in entry['groups']:
+                user_roles.extend(group_role_info.get(group_pk, []))
+
+            entry['roles'] = list(set(user_roles))
```

Comment:
We can probably have `user_roles` as set initially

Reply:
Good idea